### PR TITLE
feat(userconfig): rich description

### PR DESCRIPTION
**Pull request by Serpentiel**
## About this change—what it does

adds rich description for generated fields

## Why this way

to improve documentation

### CODE REVIEW

File: docs/resources/kafka.md [hunk 1]
```diff
@@ -100,71 +100,71 @@ Optional:
...
-- `schema_registry` (Boolean) Enable Schema-Registry service
-- `schema_registry_config` (Block List, Max: 1) Schema Registry configuration (see [below for nested schema](#nestedblock--kafka_user_config--schema_registry_config))
-- `static_ips` (Boolean) Use static public IP addresses
+- `additional_backup_regions` (List of String) Additional Cloud Regions for Backup Replication.
+- `custom_domain` (String) Serve the web frontend using a custom CNAME pointing to the Aiven DNS name.
+- `ip_filter` (List of String, Deprecated) Allow incoming connections from CIDR address block, e.g. '10.20.0.0/16'. The default value is `[0.0.0.0/0]`.
+- `ip_filter_object` (Block List, Max: 1024) Allow incoming connections from CIDR address block, e.g. '10.20.0.0/16'. The default value is `[0.0.0.0/0]`. (see [below for nested schema](#nestedblock--kafka_user_config--ip_filter_object))
+- `kafka` (Block List, Max: 1) Kafka broker configuration values. The default value is `map[]`. (see [below for nested schema](#nestedblock--kafka_user_config--kafka))
```
Comment (byashimov) [inline docs/resources/kafka.md:107]: Not sure what should we put here.

File: docs/resources/opensearch.md [hunk 2]
```diff
@@ -83,34 +83,34 @@ resource "aiven_opensearch" "os1" {
...
-- `max_index_count` (Number, Deprecated)
-- `opensearch` (Block List, Max: 1) OpenSearch settings (see [below for nested schema](#nestedblock--opensearch_user_config--opensearch))
-- `opensearch_dashboards` (Block List, Max: 1) OpenSearch Dashboards settings (see [below for nested schema](#nestedblock--opensearch_user_config--opensearch_dashboards))
-- `opensearch_version` (String) OpenSearch major version
-- `private_access` (Block List, Max: 1) Allow access to selected service ports from private networks (see [below for nested schema](#nestedblock--opensearch_user_config--private_access))
-- `privatelink_access` (Block List, Max: 1) Allow access to selected service components through Privatelink (see [below for nested schema](#nestedblock--opensearch_user_config--privatelink_access))
+- `max_index_count` (Number, Deprecated) **DEPRECATED: Usage of this field is discouraged.** use index_patterns instead. The default value is `0`.
```
Comment (byashimov) [inline docs/resources/opensearch.md:94]: It is so deprecated. Do we need warn it so many times in a row?

File: docs/resources/pg.md [hunk 3]
```diff
@@ -115,165 +115,165 @@ Read-Only:
...
-- `migration` (Block List, Max: 1) Migrate data from existing server (see [below for nested schema](#nestedblock--pg_user_config--migration))
-- `pg` (Block List, Max: 1) postgresql.conf configuration values (see [below for nested schema](#nestedblock--pg_user_config--pg))
-- `pg_read_replica` (Boolean, Deprecated)
+- `enable_ipv6` (Boolean) Register AAAA DNS records for the service, and allow IPv6 packets to service ports.
+- `ip_filter` (List of String, Deprecated) Allow incoming connections from CIDR address block, e.g. '10.20.0.0/16'. The default value is `[0.0.0.0/0]`.
+- `ip_filter_object` (Block List, Max: 1024) Allow incoming connections from CIDR address block, e.g. '10.20.0.0/16'. The default value is `[0.0.0.0/0]`. (see [below for nested schema](#nestedblock--pg_user_config--ip_filter_object))
+- `migration` (Block List, Max: 1) Migrate data from existing server. (see [below for nested schema](#nestedblock--pg_user_config--migration))
+- `pg` (Block List, Max: 1) postgresql.conf configuration values. (see [below for nested schema](#nestedblock--pg_user_config--pg))
+- `pg_read_replica` (Boolean) This setting is deprecated. Use read_replica service integration instead.
```
Comment (byashimov) [inline docs/resources/pg.md:128]: `Deprecated` is lost.

File: docs/resources/pg.md [hunk 4]
```diff
@@ -115,165 +115,165 @@ Read-Only:
...
-- `pglookout` (Block List, Max: 1) PGLookout settings (see [below for nested schema](#nestedblock--pg_user_config--pglookout))
-- `private_access` (Block List, Max: 1) Allow access to selected service ports from private networks (see [below for nested schema](#nestedblock--pg_user_config--private_access))
-- `privatelink_access` (Block List, Max: 1) Allow access to selected service components through Privatelink (see [below for nested schema](#nestedblock--pg_user_config--privatelink_access))
+- `pg_stat_monitor_enable` (Boolean) Enable the pg_stat_monitor extension. Enabling this extension will cause the cluster to be restarted.When this extension is enabled, pg_stat_statements results for utility commands are unreliable. The default value is `false`.
+- `pg_version` (String) PostgreSQL major version.
+- `pgbouncer` (Block List, Max: 1) PGBouncer connection pooling settings. (see [below for nested schema](#nestedblock--pg_user_config--pgbouncer))
+- `pglookout` (Block List, Max: 1) PGLookout settings. The default value is `map[max_failover_replication_time_lag:60]`. (see [below for nested schema](#nestedblock--pg_user_config--pglookout))
```
Comment (byashimov) [inline docs/resources/pg.md:133]: Probably must be a json like. Or not displayed at all.

File: docs/resources/pg.md [hunk 5]
```diff
@@ -115,165 +115,165 @@ Read-Only:
...
+- `wal_writer_delay` (Number) WAL flush interval in milliseconds. Note that setting this value to lower than the default 200ms may negatively impact performance.
 
 
 <a id="nestedblock--pg_user_config--pgbouncer"></a>
 ### Nested Schema for `pg_user_config.pgbouncer`
 
 Optional:
 
-- `autodb_idle_timeout` (Number) If the automatically created database pools have been unused this many seconds, they are freed. If 0 then timeout is disabled. [seconds]
+- `autodb_idle_timeout` (Number) If the automatically created database pools have been unused this many seconds, they are freed. If 0 then timeout is disabled. [seconds].
```
Comment (byashimov) [inline docs/resources/pg.md:230]: Better loose brackets.

File: docs/data-sources/account_team_member.md [hunk 6]
```diff
@@ -25,8 +25,8 @@ data "aiven_account_team_member" "foo" {
 
 ### Required
 
-- `account_id` (String) The unique account id This property cannot be changed, doing so forces recreation of the resource.
-- `team_id` (String) An account team id This property cannot be changed, doing so forces recreation of the resource.
+- `account_id` (String) The unique account id. This property cannot be changed, doing so forces recreation of the resource.
+- `team_id` (String) An account team id. This property cannot be changed, doing so forces recreation of the resource.
```
Comment (byashimov) [inline docs/data-sources/account_team_member.md:29]: Please remove trailing dot. That is 80% of diff. Let's make it in a separate PR and focus on default values now.

**Review by byashimov — CHANGES_REQUESTED** [review]

File: docs/resources/pg.md (hunk 5 above)
Comment (Serpentiel) [reply docs/resources/pg.md:230]: will be done in a separate PR

**Comment by byashimov** [comment]
LGTM.

**Review by byashimov — APPROVED** [review]

**Pull request merged**